### ISSUE TITLE
Display location permission dialog only when window.geofence.initialize() is called

### ISSUE THREAD

**Issue by shtaif** [cowbell/cordova-plugin-geofence#85]
It is quite desirable to be able to have control on when to request this permission from the user, rather than showing it as soon as the app launched for the first time.

**Comment by garrettlangley** [comment]
@shtaif did you figure out how to do this?

**Comment by shtaif** [comment]
@garrettlangley 
I do not know Swift lang, I improvised a work around that postpones the permission dialog up until you add a region to monitor for the first time (using window.geofence.addOrUpdate()), if that helps you :)

What I did is edit the GeofencePlugin.swift file inside the plugin as follows:

1. Find class "GeoNotificationManager".

2. In it's "init" method you'll find the line:
        if iOS8 {
            locationManager.requestAlwaysAuthorization()
        }
Move it into the start of the method "addOrUpdateGeoNotification", which is just the next method on this class.

3. Make this change to be picked up by the next builds you do:
Either perform the edit also in the internal copy of the GeofencePlugin.swift file in your Cordova project -> platforms -> ios -> project name -> plugins,
or else, if possible,
after editing the one in the standard Cordova project's "plugins" folder, do:
"cordova platform rm ios",
then:
"cordova platform add ios",
and it should be ready.

Hope my instructions are good for you :)
good luck.
